src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,182 @@
 import json
 import logging
+from enum import Enum
+from typing import Dict, List
+
+from d3a_interface.dataclasses import BidOfferMatch
 
 import d3a.constants
-from d3a.d3a_core.exceptions import InvalidBidOfferPair
+from d3a.d3a_core.exceptions import (
+    InvalidBidOfferPairException, OfferNotFoundException,
+    BidNotFoundException, MycoValidationException)
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
-from d3a.models.market import validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import (
-    BidOfferMatch, offer_or_bid_from_json_string)
+from d3a.models.market import Market
+
 
-from d3a.models.myco_matcher.base_matcher import BaseMatcher
+class ExternalMatcherEventsEnum(Enum):
+    OFFERS_BIDS_RESPONSE = "offers_bids_response"
+    MATCH = "match"
+    TICK = "tick"
+    MARKET = "market"
```

Comment:
I propose to be consistent.
```suggestion
    MARKET = "market_cycle"
```